client/client-lib/src/utils.rs

```diff
@@ -21,10 +22,30 @@ pub fn from_hex<D: Decodable>(s: &str) -> Result<D, anyhow::Error> {
     Ok(D::consensus_decode(std::io::Cursor::new(bytes))?)
 }
 
-pub fn parse_bitcoin_amount(
+pub fn parse_bitcoin_amount(s: &str) -> Result<bitcoin::Amount, BitcoinAmount::ParseAmountError> {
+    let maybe_i = s.find(char::is_alphabetic);
+    if let Some(i) = maybe_i {
+        //&s[0..i] is the amount
+        //&s[i..] is the denomination
+        bitcoin::Amount::from_str_in(&s[0..i], (&s[i..] as &str).parse()?)
+    } else {
+        //default to SATOSHI
+        bitcoin::Amount::from_str_in(s, bitcoin::Denomination::Satoshi)
+    }
```

Comment:
We already rely on regex, you might want to just use that, although this situation is borderline because it's so simple.

Reply:
Yes I think this dosn't really need regex. @Maan2003 suggested to use `str_split_at` and also I can get rid of the `maybe_i` and just match immediately